File: InsertionSort.java
import java.io.*;
public class InsertionSort {
	public static void main(String[] args) {
		int[] input = { 4, 2, 9, 6, 23, 12, 34, 0, 1 };
		for (int i = 1; i < input.length; i++) {
			int temp = input[i],j;
			for (j = i-1; j >= 0 && temp < input[j]; j--) {
				input[j+1] = input[j];
			}
			input[j+1] = temp;
		}

		// Print all the array elements
      for (int i = 0; i < input.length; i++) {
         System.out.println(input[i] + " ");
      }
	}
}

File: insertion_sort.py
def insertion_sort(A):
	for i in range(len(A)):
		temp = A[i]
		j = i
		while temp < A[j-1] and j > 0:
			A[j] = A[j-1]
			j -= 1
		A[j] = temp
	return A

print insertion_sort([22,1,5,9,22,7,10,44,1])
